fpmsyncd/routesync.cpp

```diff
@@ -547,6 +810,606 @@ void RouteSync::onEvpnRouteMsg(struct nlmsghdr *h, int len)
+        return;
+    }
+
+    if (!(action_str.compare("end.dt6")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.DT6 IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("end.dt4")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.DT4 IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("end.dt46")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.DT46 IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("udt6")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid uDT6 IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("udt4")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid uDT4 IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("udt46")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid uDT46 IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("end.t")) && vrf_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.T IP Prefix: %s vrf is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("end.x")) && adj_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.X IP Prefix: %s adj is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("end.dx6")) && adj_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.DX6 IP Prefix: %s adj is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    if (!(action_str.compare("end.dx4")) && adj_str.empty())
+    {
+        SWSS_LOG_NOTICE("Localsid End.DX4 IP Prefix: %s adj is empty",
+                        dstaddr_str);
+        return;
+    }
+
+    char my_sid_table_key[MAX_ADDR_SIZE + 4] = {0};
+
+    snprintf(my_sid_table_key, sizeof(my_sid_table_key), "%s", dstaddr_str);
+
+    vector<FieldValueTuple> fvVector;
+    FieldValueTuple block_len("block_len", block_len_str);
+    FieldValueTuple node_len("node_len", node_len_str);
+    FieldValueTuple func_len("func_len", func_len_str);
+    FieldValueTuple arg_len("arg_len", arg_len_str);
+    fvVector.push_back(block_len);
+    fvVector.push_back(node_len);
+    fvVector.push_back(func_len);
+    fvVector.push_back(arg_len);
+    FieldValueTuple act("action", action_str);
+    fvVector.push_back(act);
+    if (!vrf_str.empty())
+    {
+        FieldValueTuple vrf("vrf", vrf_str);
+        fvVector.push_back(vrf);
+    }
+    if (!adj_str.empty())
+    {
+        FieldValueTuple adj("adj", adj_str);
+        fvVector.push_back(adj);
+    }
+
+    m_srv6LocalSidTable.set(my_sid_table_key, fvVector);
```

Comment:
srv6orch.cpp says
    /* Key for mySid : block_len:node_len:function_len:args_len:sid-ip */

here looks like only sid-ip is used as my_sid_table_key?

Reply:
Exactly! I fixed it.